core/src/main/java/hudson/FilePath.java

```diff
@@ -984,12 +984,16 @@ private boolean installIfNecessaryFrom(@NonNull URL archive, @NonNull TaskListen
         try {
             FilePath timestamp = this.child(".timestamp");
             long lastModified = timestamp.lastModified();
+            String etag = timestamp.exists() ? timestamp.readToString().replace("\"", "") : null;
```

Comment:
This could use Util.fixEmptyAndTrim to always make it null. This avoids the double check for null and empty at later points in time

Reply:
Included in df05da5bb5bfde52278bb25f27f77f894a1fd2eb .  Thanks for review.  Could you review again?